sdk/servicebus/service-bus/src/receiver.ts

```diff
@@ -74,12 +74,18 @@ export class Receiver {
     if (!onError || typeof onError !== "function") {
       throw new Error("'onError' is a required parameter and must be of type 'function'.");
     }
-    const sReceiver = StreamingReceiver.create(this._context, {
+    StreamingReceiver.create(this._context, {
       ...options,
       receiveMode: this._receiveMode
-    });
-    this._context.streamingReceiver = sReceiver;
-    return sReceiver.receive(onMessage, onError);
+    })
+      .then((sReceiver) => {
+        if (!this.isClosed && this._context.streamingReceiver) {
+          return sReceiver.receive(onMessage, onError);
+        }
+      })
+      .catch((err) => {
+        onError(err);
+      });
```

Comment:
Since `StreamingReceiver.create` is no longer returning a promise, the changes in this section should be reverted

Reply:
It seems like these changes are necessary to stay in sync with how we are doing things in messageSession? (Error handling and setting up messageHandler after receiver is created?)

Reply:
It seems like these changes are necessary to stay in sync with how we are doing things in messageSession? (Error handling and setting up messageHandler after receiver is created?)

Reply:
> > About the issue with moving init(), the normal scenario of receiving messages was not working (no messages were being received at all, even when adding delay and closing after the await). Hence, tried out the other alternative we talked about, to intercept the addCredit/listeners setup after or during init().
> 
> I tried moving the `init()` call from `receive()` to `create()` in the Streaming Receiver, and the tests for the streaming receiver work just fine
> 
> For reference: This [commit](https://github.com/ramya-rao-a/azure-sdk-for-js/commit/47038053e1695cc8b5d41f77fd35f3363cda8b2d) has my changes and your tests. I ran all the streaming receiver tests for the unpartitioned queue, and they all pass.

Got it, the isOpen() check removal was necessary too for the move. Factored this in in the latest changes.

Reply:
Moved _init() to create. Refer to https://github.com/Azure/azure-sdk-for-js/pull/2139#issuecomment-482215941